### PR TITLE
Implement unsaved change indicator

### LINKED ISSUE #285: [Portal] Unsaved changes indicator
Indicates fields with unsaved changes.

ref #243

### PR DESCRIPTION
**Pull request by axolotl0212**
fix #285 

# Screenshot
Screenshot 2020-10-16 at 11 26 40 PM

### CODE REVIEW

File: portal/src/graphql/adminapi/ResetPasswordScreen.tsx [hunk 1]
```diff
@@ -58,25 +63,25 @@ const ResetPasswordForm: React.FC<ResetPasswordFormProps> = function (
...
+    setFormData((prev) => ({ ...prev, newPassword: value }));
+  });
+  const { onChange: onConfirmPasswordChange } = useTextField((value) => {
+    setFormData((prev) => ({ ...prev, confirmPassword: value }));
+  });
 
   const isFormModified = useMemo(() => {
-    return !deepEqual({ newPassword: "", confirmPassword: "" }, screenState);
-  }, [screenState]);
+    return !deepEqual({ newPassword: "", confirmPassword: "" }, formData);
```
Comment (louischan-oursky) [inline portal/src/graphql/adminapi/ResetPasswordScreen.tsx]: Use `initialFormData` here.

File: portal/src/graphql/portal/AuthenticationAuthenticatorSettings.tsx [hunk 2]
```diff
@@ -200,6 +204,21 @@ function getActivatedKeyListFromState<KeyType>(
...
+  >,
+  field: K,
+  action: React.SetStateAction<AuthenticationAuthenticatorScreenState[K]>
+) {
+  setState((prev) => ({
+    ...prev,
+    [field]: typeof action === "function" ? action(prev[field]) : action,
+  }));
+}
+
```
Comment (louischan-oursky) [inline portal/src/graphql/portal/AuthenticationAuthenticatorSettings.tsx:222]: This function seems a bit redundant. Why don't we just use `setState` inline?

File: portal/src/graphql/portal/AuthenticationLoginIDSettings.tsx [hunk 3]
```diff
@@ -74,6 +74,21 @@ const widgetTitleMessageId: Record<LoginIDKeyType, string> = {
...
+  >,
+  field: K,
+  action: React.SetStateAction<AuthenticationLoginIDSettingsState[K]>
+) {
+  setState((prev) => ({
+    ...prev,
+    [field]: typeof action === "function" ? action(prev[field]) : action,
+  }));
+}
+
```
Comment (louischan-oursky) [inline portal/src/graphql/portal/AuthenticationLoginIDSettings.tsx:92]: This function seems a bit redundant. Why don't we just use `setState` inline?

File: portal/src/graphql/portal/AuthenticationAuthenticatorSettings.tsx (hunk 2 above)
Comment (axolotl0212) [reply portal/src/graphql/portal/AuthenticationAuthenticatorSettings.tsx:222]: Let's change to using `setState` inline. I abstract this to a function as it seems repetitive

**Comment by axolotl0212** [comment]
@louischan-oursky updated

**Pull request merged**